### PR TITLE
MDEV-28153: Debian autobake- use absolute dependencies rather than a buildtime detection (10.6)

### PR DESCRIPTION
**Pull request by grooverdan**
- [x] *The Jira issue number for this PR is: MDEV-28153*

## Description

like #2055

## How can this PR be tested?

https://salsa.debian.org/mariadb-team/mariadb-server/-/merge_requests/9

## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature and the PR is based against the latest MariaDB development branch*
- [ X ] *This is a bug fix and the PR is based against the earliest branch in which the bug can be reproduced*

### CODE REVIEW

**Comment by illuusio** [comment]
What do you mean with static dependencies?

**Comment by ottok** [comment]
Good catch riscv64 also supports pmem now! I will test that downstream Debian next week.

File: debian/autobake-deb.sh [hunk 1]
```diff
@@ -50,40 +50,82 @@ remove_rocksdb_tools()
...
+    disable_pmem
+    ;&
+  buster)
+    replace_uring_with_aio
+    if [ ! "$architecture" = amd64 ]
+    then
+      disable_pmem
+    fi
+    ;&
+  bullseye|bookworm)
```
Comment (ottok) [inline debian/autobake-deb.sh:84]: For the time being, bookworm=sid and should just work out-of-the-box.

I still think using `apt-cache madison` was a good system and apt-cache to my knowledge always works if the build host is healthy.

File: debian/autobake-deb.sh [hunk 2]
```diff
@@ -50,40 +50,82 @@ remove_rocksdb_tools()
...
+    ;&
+  impish|jammy)
+    # mariadb-plugin-rocksdb s390x not supported by us (yet)
+    # ubuntu doesn't support mips64el yet, so keep this just
+    # in case something changes.
+    if [[ ! "$architecture" =~ amd64|arm64|ppc64el|s390x ]]
+    then
+      remove_rocksdb_tools
+    fi
+    if [[ ! "$architecture" =~ amd64|arm64|ppc64el ]]
```
Comment (ottok) [inline debian/autobake-deb.sh:116]: Latest version of this PR no longer has riscv64 here, so I assume it didn't work on that arch.

**Review by ottok — CHANGES_REQUESTED** [review]

**Comment by ottok** [comment]
The branch on Debian Salsa: https://salsa.debian.org/grooverdan/mariadb-server/-/tree/10.6-MDEV-28011-test-pmem-autobake
Salsa-CI passes on everything apart from the build-all/build-any cross-build tests, but that is unrelated.

**Comment by ottok** [comment]
Looking at https://buildd.debian.org/status/package.php?p=pmdk riscv64 is supported, so I will test that in next Debian upload of mariadb-10.6.

**Comment by ottok** [comment]
> apt-cache madison was decent, but we don't build with hosts any more, only containers. Clearing cache on containers is pretty standard.

Sure you can clean the apt cache but you need to load it anyway before installing anything (e.g. `apt update`), otherwise apt will not work at all, hence `apt-cache madison` will always work if `apt install` works and builder is healthy.

**Comment by grooverdan** [comment]
closed via 1b2ee693b7cf2f514fa93c3acc047d4328b96364
also added pmem on riscv64 63f76d3b98ea5bc829c0bc08830ec3a9267915c4

**Comment by ottok** [comment]
My review feedback was "request changes" - those were ignored and instead multiple extra commits added and then merged straight away. This isn't how I would expect a Pull Request review to go.

**Pull request merged**